lib/capabilities.js

```diff
@@ -17,21 +18,177 @@ var async = require('async');
+            cb(true);
+          } else {
+            cb(false, 'Tracing license is missing or expired.');
+          }
+        }
+      },
+      {
+        check: function(cb) {
+          if (process.env.STRONGLOOP_TRACING) {
+            cb(true);
+          } else {
+            cb(false, 'Tracing is disabled.');
+          }
+        }
+      }
+    ]
+  },
+  metrics: {
+    name: 'Metrics',
+    checks: [
+      {
+        check: function(cb) {
+          if (license('agent:metrics')) {
+            cb(true);
+          } else {
+            cb(false, 'Metrics license is missing or expired.');
+          }
+        }
+      },
+      {
+        check: function(cb) {
+          if (agent().metrics) {
+            cb(true);
+          } else {
+            cb(false, 'Metrics is disabled.');
+          }
+        }
+      },
+      {
+        check: function(cb) {
+          var addon = require('strong-agent/lib/addon');
+          if (addon) {
+            cb(true);
+          } else {
+            cb(false, 'Strong-agent addon not compiled.');
+          }
+        }
+      }
+    ]
+  },
+  cpuprofile: {
+    name: 'CPU profiling',
+    checks: [
+      {
+        check: function(cb) {
+          if (license('agent:watchdog')) {
+            cb(true);
+          } else {
+            cb(false, 'Smart profiling license is missing or expired.');
+          }
+        }
+      },
+      {
+        check: function(cb) {
+          var addon = require('strong-agent/lib/addon');
+          if (addon) {
+            cb(true);
+          } else {
+            cb(false, 'Strong-agent addon not compiled.');
+          }
+        }
+      },
+      {
+        check: function(cb) {
+          var platform = require('os').platform();
+          if (isSupported(platform)) {
+            cb(true);
+          } else {
+            cb(false, 'Host platform does not support CPU profiling.');
+          }
+        }
+      }
+    ]
+  },
+  heapsnapshot: {
+    name: 'Heap snapshot',
+    checks: [
+      {
+        check: function(cb) {
+          var heapdump = require('heapdump');
+          if (heapdump) {
+            cb(true);
+          } else {
+            cb(false, 'Heap snapshot not compiled.');
+          }
+        }
+      },
+      {
+        check: function(cb) {
+          var platform = require('os').platform();
+          if (isSupported(platform)) {
+            cb(true);
+          } else {
+            cb(false, 'Host platform does not support heap snapshot.');
+          }
+        }
+      }
+    ]
+  },
+  patch: {
+    name: 'patch',
+    checks: [
+      {
+        check: function(cb) {
+          if (agent().dyninst.metrics) {
+            cb(true);
+          } else {
+            cb(false,
+              'Agent is not configured to support dynamic instrumentation.');
+          }
+        }
+      }
+    ]
+  },
 };
 
+// linux, sunos, solaris, darwin, freebsd, win32,
+function isSupported(platform) {
```

Comment:
This check appears to list every OS supported by node and is used by multiple features.. this check would only be meaningful if the feature itself was tied to a platform and the check only checked for those specific platforms..